hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/FullScanPhysicalRule.java

```diff
@@ -46,4 +57,38 @@ public RelNode convert(RelNode rel) {
                 logicalScan.getTable()
         );
     }
+
+    @Override
+    public void onMatch(RelOptRuleCall call) {
```

Comment:
If you need to override `onMatch()` most probably the rule should simply extend `RelOptRule` not `ConverterRule`.

Reply:
IMO, no need to do that, it's only overcomplicate the implementation.

Reply:
It actually simplifies it, `RelOptRule` is more generic.